src/traces/scatter/index.js

```diff
@@ -855,3 +857,53 @@ scatter.hoverPoints = function(pointData, xval, yval, hovermode) {
 
     return [pointData];
 };
+
+var DESELECTDIM = 0.2;
+
+scatter.selectPoints = function(searchInfo, polygon) {
+    var cd = searchInfo.cd,
+        xa = searchInfo.xaxis,
+        ya = searchInfo.yaxis,
+        selection = [],
+        curveNumber = cd[0].trace.index,
+        marker = cd[0].trace.marker,
+        i,
+        di,
+        x,
+        y;
+
+    if(!marker) return; // TODO: include text and/or lines?
+
+    var opacity = Array.isArray(marker.opacity) ? 1 : marker.opacity;
+
+    if(polygon === false) { // clear selection
+        for(i = 0; i < cd.length; i++) cd[i].dim = 0;
+    }
+    else {
+        for(i = 0; i < cd.length; i++) {
+            di = cd[i];
+            x = xa.c2p(di.x);
+            y = ya.c2p(di.y);
+            if(polygon.contains([x, y])) {
+                selection.push({
+                    curveNumber: curveNumber,
+                    pointNumber: i,
+                    x: di.x,
+                    y: di.y
+                });
```

Comment:
> If rectangular, will we get the selection box boundaries too?

The tricky thing here is what to do with overlaid axes. I suppose we could report things like `xmin`, `x2min`, `x3min`... during the rest of the process I do all the calculations in pixels not data coordinates so this isn't an issue, but if we settle on a structure it would be easy to include the data coordinates in the event data.

Which then raises another tricky point about persisting selections when you zoom/pan, since you can drag on an individual axis and change `x` independently from `x2` - I guess the selected *points* would separate (that would be cool to watch!) but what then would you do with the selection outline? That can be sorted out in V2 I guess.